geode-core/src/main/java/org/apache/geode/cache/client/internal/ExecuteRegionFunctionOp.java

```diff
@@ -310,29 +198,22 @@ public static void reexecute(ExecutablePool pool, String region, String function
 
     private FunctionException functionException;
 
-
-    public ExecuteRegionFunctionOpImpl(String region, Function function,
-        ServerRegionFunctionExecutor serverRegionExecutor, ResultCollector rc, byte hasResult,
+    private static int getMsgParts(int filterSize,
         Set<String> removedNodes) {
-      super(MessageType.EXECUTE_REGION_FUNCTION,
-          8 + serverRegionExecutor.getFilter().size() + removedNodes.size());
+      return 8 + filterSize + removedNodes.size();
+    }
+
+    private void fillMessage(String region, Object function,
```

Comment:
This is just a comment but I wanted you to consider not making the "function" parameter's type "Object". I think I see why you did it (one caller exists that just gave us a String as the function).
I'd prefer that fillMessage have two parameters: one typed as Function and the other String (call it functionId). This allows the call of serverRegionExecutor.isFnSerializationReqd() to still be here instead of in multiple callers.
I think you would just need to also test for "function" being not-null:
if (function != null && serverRegionExecutor.isFnSerializationReqd()) {
  getMessage().addStringOrObjPart(function);
} else {
  getMessage().addStringOrObjPart(functionId);
}
Other refactorings than this might also exist. I didn't like losing the type information and I didn't like the test of isFnSerializationReqd being in multiple places.

Reply:
I agree with you. Thanks.